code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/MessageWebViewRunner.java

```diff
@@ -295,26 +307,44 @@ private void createMessageFrameAndAddMessageToRootView(final MessageSettings set
             webviewSettings.setUseWideViewPort(true);
         }
 
-        // create a new view to apply a background dimming effect behind a displayed message
-        backdrop =
-                new View(
-                        ServiceProvider.getInstance()
-                                .getAppContextService()
-                                .getApplicationContext());
-        backdrop.setLayoutParams(
-                new LayoutParams(LayoutParams.MATCH_PARENT, LayoutParams.MATCH_PARENT));
-        backdrop.setBackgroundColor(Color.parseColor(settings.getBackdropColor()));
-        backdrop.setAlpha(settings.getBackdropOpacity());
-
-        // add the webview overlaid on the backdrop
-        message.rootViewGroup.addView(
-                backdrop, message.baseRootViewWidth, message.baseRootViewHeight);
-        message.rootViewGroup.addView(webView, params);
+        // apply round corners to a CardView then add the webview
+        final Context context =
+                ServiceProvider.getInstance().getAppContextService().getApplicationContext();
+        webViewFrame = new CardView(context);
+        final float calculatedRadius =
+                TypedValue.applyDimension(
+                        TypedValue.COMPLEX_UNIT_DIP,
+                        settings.getCornerRadius(),
+                        context.getResources().getDisplayMetrics());
+        webViewFrame.setRadius(calculatedRadius);
+        webViewFrame.addView(message.webView);
```

Comment:
Is CardView being used only for the ability to set the radius? If so, have you already tried to change the shape of the layout programmatically applying a shape(something like [this](https://github.com/adobe/aepsdk-assurance-android/blob/dev/code/assurance/src/main/res/drawable/shape_custom_button_filled.xml#L16) but in code) ? I ask because, adding the CardView dependency will require implementers that do not need Messaging to pull this dependency. 

If there is no other way to apply this rounded corners, then we should add this to the [implicit dependency in pom as well](https://github.com/adobe/aepsdk-core-android/blob/main/code/core/release.gradle#L51)

Reply:
I did try applying a rounded rectangle as a background for the webview but it was not working as desired as the webview would retain the shape specified in the html content.

Reply:
Just to clarify - did you also try wrapping the webview inside a FrameLayout and then applying the shape to the Frame layout?

Reply:
i was applying the shape as a background to the webview. i just tried applying the shape as a background to a FrameLayout then added the webview as a child view which then gave the same incorrect behavior as I saw before.

Reply:
Got it. Thank you for testing that Ryan! We recently had some requests around removing view dependencies as much as possible in core/assurance. So just wanted to make sure we explored alternatives if they exists.